### DISCUSSION TITLE
Undefined property: stdClass::$id

### DISCUSSION THREAD

**Discussion by puntogestioncl** [rappasoft/laravel-livewire-tables#1070] [Q&A] [answered]
Hello,

Excuse my English, I am using the translator, since it is low level.
I have a little problem, when selecting all in bulk actions. it gives me an error:

`Undefined property: stdClass::$id`

In my case I have a detail of invoices, therefore, my ID is the number of this, and not that of the detail, so in 

`$this->setPrimaryKey('invoice.id')`

So it forces me to add the id column:

`Column::make("ID", "id")->sortable(),`

and change to:

`$this->setPrimaryKey('id')`

so as not to throw the error.

This error does not appear when selecting only some rows.

Hopefully they can help me. 

Greetings.

**Comment by lrljoe** [comment, marked as answer]
Use the following function to add the ID into the query.
```php
public function configure(): void { 
$this->setAdditionalSelects(['users.id as id']); 
}
```

**Reply by puntogestioncl** [reply]
Thank you very much, it worked.
